src/software/sensor_fusion/sensor_fusion.h

```diff
@@ -48,13 +48,32 @@ class SensorFusion
      */
     std::optional<World> getWorld() const;
 
+    /**
+     * This function controls the behavior of how the ball is being updated. If this
+     * returns True, we use the position of the robot that triggers the breakbeam instead
+     * of the one given by the ssl vision system.
+     *
+     * @return True if we were to use the position of the robot instead of the ssl camera
+     * system. False otherwise
+     */
+    bool shouldUseRobotBallPositionInsteadOfVision(
+        const SSLProto::SSL_DetectionFrame &frame,
+        const std::vector<BallDetection> &ball_detection);
```

Comment:
I think this is more appropriate as a private function

Reply:
Yes, I do also think so. 

The problem with is that I can't test this function. I am not quite sure how you friend this class for gtest.

Reply:
Nvm, I  just saw the comment below.